src/util/RtreeFileReader.cpp

```diff
@@ -0,0 +1,78 @@
+//  Copyright 2023, University of Freiburg,
+//                  Chair of Algorithms and Data Structures.
+//  Author: Noah Nock <noah.v.nock@gmail.com>
+
+#include <util/Rtree.h>
+#include <util/RtreeFileReader.h>
+
+FileReader::FileReader(const std::string& filePath) {
```

Comment:
Where is the corresponding "writer" to this "reader", 
I think they should naturally be put together, then you can also simplify the code.